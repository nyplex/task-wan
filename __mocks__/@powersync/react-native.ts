module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
