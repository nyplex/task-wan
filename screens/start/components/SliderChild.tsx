import { Image } from "expo-image";
import { Box } from "@/components/gluestack/box";
import Text from "@/components/Text";

type Props = {
  title: string;
  imageSource: string;
  description: string;
};

const SliderChild = ({ title, imageSource, description }: Props) => {
  return (
    <Box className="px-4">
      <Box>
        <Image
          source={imageSource}
          contentFit="contain"
          style={{ width: "100%", height: 400 }}
        />
      </Box>
      <Box className="w-full h-full px-4">
        <Text size="body" weight="semi-bold" className="text-center">
          {title}
        </Text>
        <Text size="bodyS" className="text-center mt-2">
          {description}
        </Text>
      </Box>
    </Box>
  );
};

export default SliderChild;
