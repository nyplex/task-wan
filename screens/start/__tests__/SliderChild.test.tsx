import { render } from "@testing-library/react-native";
import SliderChild from "../components/SliderChild";
import React from "react";

describe("SliderChild", () => {
  it("renders title, description, and image", () => {
    const { getByText } = render(
      <SliderChild
        title="Test Title"
        description="Test Description"
        imageSource="test.png"
      />
    );
    expect(getByText("Test Title")).toBeTruthy();
    expect(getByText("Test Description")).toBeTruthy();
  });

  it("matches the rendered output snapshot", () => {
    const { toJSON } = render(
      <SliderChild
        title="Snapshot Title"
        description="Snapshot Description"
        imageSource="snapshot.png"
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
