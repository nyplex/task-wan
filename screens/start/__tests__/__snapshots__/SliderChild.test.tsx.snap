// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderChild matches the rendered output snapshot 1`] = `
<View
  className="px-4"
>
  <View>
    <ViewManagerAdapter_ExpoImage
      containerViewRef={"[React.ref]"}
      contentFit="contain"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height={400}
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={[]}
      source={
        [
          {
            "uri": "snapshot.png",
          },
        ]
      }
      style={
        {
          "height": 400,
          "width": "100%",
        }
      }
      transition={null}
      width="100%"
    />
  </View>
  <View
    className="w-full h-full px-4"
  >
    <Text
      className="font-body based-styles font-semibold text-typography-primary text-[16px] tracking-[0px] text-center"
    >
      Snapshot Title
    </Text>
    <Text
      className="based-styles font-regular text-typography-primary text-[14px] tracking-[0px] text-center mt-2"
    >
      Snapshot Description
    </Text>
  </View>
</View>
`;
